feat: add `in_pkgdown_pkg()`  and `IN_PKGDOWN_PKG` (#2977)
r-lib/pkgdown: NEWS.md

* do not autolink code that is in a link (href) in Rd files (#2972)

* New `in_pkgdown_pkg()` function and new `IN_PKGDOWN_PKG` variable, to go with 

> `in_pkgdown()` and `IN_PKGDOWN`, but specific to the package whose site is being built (#2976). 

Reviewer: Nice, thanks!

```suggestion
  `in_pkgdown()` and `IN_PKGDOWN`. Returns/contains the name of the package
  whose site is being built (#2976). 
```
